src/lib.rs

```diff
@@ -16,12 +17,19 @@ pub mod inc_encoding;
 pub mod signature;
 pub mod symmetric;
 
+/// Cached Poseidon2 permutations.
+///
+/// We cache the default Plonky3 Poseidon2 instances once and return a clone.
+/// Returning by value preserves existing call sites that take `&perm`.
+static POSEIDON2_24: OnceLock<Poseidon2BabyBear<24>> = OnceLock::new();
+static POSEIDON2_16: OnceLock<Poseidon2BabyBear<16>> = OnceLock::new();
```

Comment:
Just so that we can view the header documentation independently for each instance :) 

```suggestion
// Cached Poseidon2 permutations. 
//
// We cache the default Plonky3 Poseidon2 instances once and return a clone.
// Returning by value preserves existing call sites that take `&perm`.

/// A lazily-initialized, thread-safe cache for the Poseidon2 permutation with a width of 24.
static POSEIDON2_24: OnceLock<Poseidon2BabyBear<24>> = OnceLock::new();

/// A lazily-initialized, thread-safe cache for the Poseidon2 permutation with a width of 16.
static POSEIDON2_16: OnceLock<Poseidon2BabyBear<16>> = OnceLock::new();
```